src/components/multi_select_state_utils.js

```diff
@@ -1,18 +1,36 @@
-export const findItemByIds = ids => item =>
-  ids.find(id => id === item.id) === undefined;
+const matchItemById = id => item => id === item.id;
 
-export const getSelectAllItems = (filteredItems, selectedItems, items) => {
-  const sourceItems = items.filter(
+const getItemById = (id, selectedItems) =>
+  selectedItems.find(matchItemById(id));
```

Comment:
i think no need to move "item => id === item.id" to separate method, lets keep it here